core/src/main/java/org/mule/runtime/core/api/config/DefaultMuleConfiguration.java

```diff
@@ -600,6 +604,21 @@ public void setDefaultObjectSerializer(ObjectSerializer defaultObjectSerializer)
     this.defaultObjectSerializer = defaultObjectSerializer;
   }
 
+  /**
+   * @return the name of the folder associated to this artifact to use for storing artifact related data
+   */
+  public String getDataFolderName() {
+    return dataFolderName != null ? dataFolderName : getId();
```

Comment:
Is not dataFolder the same as workingDirectory?

Reply:
No, it's just the name of the folder to use for temporary storage. The working directory is the whole file path (including the MULE_BASE/.mule )